ejb/src/main/java/org/wildfly/httpclient/ejb/ClientHandlers.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.httpclient.ejb;
+
+import io.undertow.client.ClientResponse;
+import org.jboss.ejb.client.EJBClientInvocationContext;
+import org.jboss.ejb.client.EJBModuleIdentifier;
+import org.jboss.ejb.client.EJBReceiverInvocationContext;
+import org.jboss.ejb.client.SessionID;
+import org.jboss.marshalling.ByteInput;
+import org.jboss.marshalling.ByteOutput;
+import org.jboss.marshalling.InputStreamByteInput;
+import org.jboss.marshalling.Marshaller;
+import org.jboss.marshalling.Marshalling;
+import org.jboss.marshalling.Unmarshaller;
+import org.wildfly.httpclient.common.HttpTargetContext;
+import org.xnio.IoUtils;
+
+import java.io.Closeable;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static org.wildfly.httpclient.ejb.ByteOutputs.byteOutputOf;
+import static org.wildfly.httpclient.ejb.Serializer.deserializeObject;
+import static org.wildfly.httpclient.ejb.Serializer.deserializeSet;
+import static org.wildfly.httpclient.ejb.Serializer.serializeMap;
+import static org.wildfly.httpclient.ejb.Serializer.serializeObjectArray;
+import static org.wildfly.httpclient.ejb.Serializer.serializeTransaction;
+import static org.wildfly.httpclient.ejb.Serializer.deserializeMap;
+import static org.xnio.IoUtils.safeClose;
+
+/**
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+final class ClientHandlers {
```

Comment:
To make things a little more orderly, I would move the static methods so that (1) the marshallers appear before the result handlers and (2)  within the marshallers, order them in the order invoke, create session, discover, cancel and similarly for the  result handlers, with a small line comment introducing each section.

Reply:
Fixed, now marshalling methods are first, result handlers are second & functions are third.